src/pipx/commands/install.py

```diff
@@ -186,41 +188,52 @@ def install_all(
 ) -> ExitCode:
     """Return pipx exit code."""
     venv_container = VenvContainer(paths.ctx.venvs)
+    failed: List[str] = []
+    installed: List[str] = []
 
     for venv_metadata in extract_venv_metadata(spec_metadata_file):
         # Install the main package
         main_package = venv_metadata.main_package
         venv_dir = venv_container.get_venv_dir(f"{main_package.package}{main_package.suffix}")
-        install(
-            venv_dir,
-            None,
-            [generate_package_spec(main_package)],
-            local_bin_dir,
-            local_man_dir,
-            python or get_python_interpreter(venv_metadata.source_interpreter),
-            pip_args,
-            venv_args,
-            verbose,
-            force=force,
-            reinstall=False,
-            include_dependencies=main_package.include_dependencies,
-            preinstall_packages=[],
-            suffix=main_package.suffix,
-        )
-
-        # Install the injected packages
-        for inject_package in venv_metadata.injected_packages.values():
-            commands.inject(
+        try:
+            install(
                 venv_dir,
                 None,
-                [generate_package_spec(inject_package)],
+                [generate_package_spec(main_package)],
+                local_bin_dir,
+                local_man_dir,
+                python or get_python_interpreter(venv_metadata.source_interpreter),
                 pip_args,
-                verbose=verbose,
-                include_apps=inject_package.include_apps,
-                include_dependencies=inject_package.include_dependencies,
+                venv_args,
+                verbose,
                 force=force,
-                suffix=inject_package.suffix == main_package.suffix,
+                reinstall=False,
+                include_dependencies=main_package.include_dependencies,
+                preinstall_packages=[],
+                suffix=main_package.suffix,
             )
 
+            # Install the injected packages
+            for inject_package in venv_metadata.injected_packages.values():
+                commands.inject(
+                    venv_dir,
+                    None,
+                    [generate_package_spec(inject_package)],
+                    pip_args,
+                    verbose=verbose,
+                    include_apps=inject_package.include_apps,
+                    include_dependencies=inject_package.include_dependencies,
+                    force=force,
+                    suffix=inject_package.suffix == main_package.suffix,
+                )
+        except PipxError as e:
+            print(e, file=sys.stderr)
+            failed.append(venv_dir.name)
+        else:
+            installed.append(venv_dir.name)
+    if len(installed) == 0:
+        print(f"No packages installed after running 'pipx install-all {spec_metadata_file}' {sleep}")
```

Comment:
Stylistic nitpick, we usually format our output like this.

```suggestion
        print(f"{sleep} No packages installed after running 'pipx install-all {spec_metadata_file}'")
```

Reply:
Done.